### PR TITLE
[PM-10682] user state subject

### PR DESCRIPTION
**Pull request by audreyality**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10682

## 📔 Objective

The user state subject adapts state providers to an rxjs subject.

The subject will be used by the generator UI to load and store the active user’s global generator settings.

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/common/src/tools/dependencies.ts
Comment (audreyality) [inline libs/common/src/tools/dependencies.ts:1]: 📝 Not all of the types in this file are used by the user-state subject, but they *are* all used by the service code in downstream PRs.

File: libs/common/src/tools/dependencies.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+/** A pattern for types that emit values exclusively when the dependency
+ *  emits a message.
+ *
+ *  Consumers of this dependency should emit when `when$` emits. If `when$`
+ *  completes, the consumer should also complete. If `when$`
+ *  errors, the consumer should also emit the error.
+ *
+ *  @remarks This dependency is useful when you have a nondeterministic
+ *  or stateful algorithm that you would like to run when an event occurs.
+ */
```
Comment (djsmith85) [inline libs/common/src/tools/dependencies.ts:76]: ⚠️  Documentation needs to be updated after re-name of the dependency

File: libs/common/src/tools/dependencies.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ *  value. If `when$` completes, the consumer should also complete. If
+ * `when$` errors, the consumer should also emit the error.
+ *
+ *  @remarks Check the consumer's documentation to determine how it
+ *  responds to emissions.
+ */
+export type WhenDependency = {
+  /** The stream to observe for true emissions. */
+  when$: Observable<boolean>;
+};
```
Comment (djsmith85) [inline libs/common/src/tools/dependencies.ts:95]: ❓ Should we name it WhenTrue, if it only emits when true?

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Nice work, which we already discussed last week. Just some documentation updates needed before we can merge

File: libs/common/src/tools/dependencies.ts (hunk 2 above)
Comment (audreyality) [reply libs/common/src/tools/dependencies.ts:95]: I don't believe that's necessary. It's common for other conditionals (`if`, `while`, rxjs `filter`, etc) that we don't call out they only execute when their value is true.

File: libs/common/src/tools/dependencies.ts (hunk 1 above)
Comment (audreyality) [reply libs/common/src/tools/dependencies.ts]: Addressed in 2e5f7d8df1d97ab202.

**Comment by audreyality** [comment]
❗ Please don't review this until I've addressed the remaining unit tests. :)

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**